core/commoncmd/daemon_hb_rotate.go

```diff
@@ -0,0 +1,150 @@
+package commoncmd
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	"github.com/opensvc/om3/core/client"
+	"github.com/opensvc/om3/daemon/msgbus"
+)
+
+type (
+	CmdDaemonHeartbeatRotate struct {
+		Wait bool
+	}
+)
+
+func NewCmdDaemonHeartbeatRotate() *cobra.Command {
+	options := CmdDaemonHeartbeatRotate{}
+	cmd := &cobra.Command{
+		Use:   "rotate",
+		Short: "rotate a disk heartbeat encryption secret",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return options.Run()
+		},
+	}
+	flags := cmd.Flags()
+	flags.BoolVar(&options.Wait, "wait", false, "wait for the rotate operation to complete")
+	return cmd
+}
+
+func (t *CmdDaemonHeartbeatRotate) Run() error {
+	timeoutDuration := 30 * time.Second
+	done := make(chan error, 1)
+
+	c, err := client.New()
+	if err != nil {
+		return err
+	}
+
+	var (
+		ctx    context.Context
+		cancel context.CancelFunc
+	)
+
+	if t.Wait {
+		ctx, cancel = context.WithTimeout(context.Background(), timeoutDuration)
+	} else {
+		ctx, cancel = context.WithCancel(context.Background())
+	}
+	defer cancel()
+
+	if t.Wait {
+		if err := startEventWatcher(ctx, c, done, timeoutDuration); err != nil {
+			return err
+		}
+	}
+
+	resp, err := c.PostClusterHeartbeatRotateWithResponse(ctx)
+	if err != nil {
+		return err
+	}
+
+	if t.Wait && resp.StatusCode() == 200 {
+		select {
+		case <-ctx.Done():
+			return fmt.Errorf("timeout or cancelled while waiting for event: %w", ctx.Err())
+		case e := <-done:
+			return e
+		}
+	}
+
+	switch resp.StatusCode() {
+	case 200:
+		if !t.Wait {
+			fmt.Printf("ID : %s\n", resp.JSON200.ID.String())
+		}
+	case 400:
+		return fmt.Errorf("%s", resp.JSON400)
+	case 401:
+		return fmt.Errorf("%s", resp.JSON401)
+	case 403:
+		return fmt.Errorf("%s", resp.JSON403)
+	case 409:
+		return fmt.Errorf("%s", resp.JSON409)
+	case 500:
+		return fmt.Errorf("%s", resp.JSON500)
+	default:
+		return fmt.Errorf("unexpected status code: %d", resp.StatusCode())
+
+	}
+
+	return nil
+}
+
+func startEventWatcher(ctx context.Context, c *client.T, done chan<- error, timeoutDuration time.Duration) error {
+	getEvents := c.NewGetEvents().
+		SetFilters([]string{"HeartbeatRotateSuccess", "HeartbeatRotateError"}).
+		SetDuration(timeoutDuration)
+
+	evReader, err := getEvents.GetReader()
+	if err != nil {
+		return err
+	}
+
+	go func() {
```

Comment:
can be moved inside the second goroutine
with a dedicated context to avoid goroutine leak if ctx never ends